op-service/txmgr/txmgr_test.go

```diff
@@ -1003,3 +1003,61 @@ func TestNonceReset(t *testing.T) {
 	// internal nonce tracking should be reset every 3rd tx
 	require.Equal(t, []uint64{0, 0, 1, 2, 0, 1, 2, 0}, nonces)
 }
+
+// TestClose ensures that the tx manager will refuse new work and cancel any in progress
+func TestClose(t *testing.T) {
```

Comment:
Does this test the case where we enter
```go
if sendState.IsWaitingForConfirmation() {
				continue
			}
```
?

Reply:
No, this test only confirms that Closing is done when the manager is closed, but doesn't test this case.

I'm including a new test, `TestCloseWaitingForConfirmation`, which has different config/setup to test this case.